src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -427,7 +429,14 @@ public async Task<IActionResult> ExternalLoginCallback(string returnUrl = null,
 
                     // Link external login to an existing user
                     ViewData["UserName"] = iUser.UserName;
-                    ViewData["Email"] = email;
+                    if (userLocator != null)
+                    {
+                        ViewData["LinkParameterValue"] = userLocator.GetValueThatLinkAccount(info);
+                    } 
+                    else 
+                    {
+                        ViewData["LinkParameterValue"] = info.GetEmail();
```

Comment:
Instead of calling it `LinkParameterValue`, can we name it `ExternalUserIdentifier` instead?

Reply:
fixed in 4726652